src/core/ai/llm/validation.ts

```diff
@@ -0,0 +1,62 @@
+import { validateModelFileSupport, validateFileSupport } from './registry.js';
+import { logger } from '../../logger/index.js';
+
+export interface FileValidationResult {
+    isSupported: boolean;
+    fileType?: string;
+    error?: string;
+}
+
+export interface FileValidationConfig {
+    provider: string;
+    model?: string;
+}
+
+/**
+ * Validates if a file is supported by the current LLM configuration.
+ * Uses model-specific validation when available, falls back to provider-level.
+ * @param config The LLM configuration containing provider and optional model
+ * @param mimeType The MIME type of the file to validate
+ * @returns Validation result with support status and error details
+ */
+export function validateFileForLLM(
```

Comment:
we can make this `validateInputForLLM` as mentioned earlier that accepts the entire API inputs